src/main/java/com/squareup/javapoet/SelfName.java

```diff
@@ -0,0 +1,26 @@
+package com.squareup.javapoet;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+public class SelfName extends TypeName {
+
+  public static SelfName get() {
+    return new SelfName(Collections.<AnnotationSpec>emptyList());
+  }
+
+  SelfName(List<AnnotationSpec> annotations) {
+    super(annotations);
+  }
+
+  @Override public TypeName annotated(List<AnnotationSpec> annotations) {
+    return new SelfName(annotations);
```

Comment:
Consider looking at/commenting on/merging #365? It'd help writing those tests (with less code). ;-)